src/Common/src/System/Globalization/IdnMapping.cs

```diff
@@ -131,7 +135,13 @@ public string GetUnicode(string ascii, int index, int count)
             // We're only using part of the string
             ascii = ascii.Substring(index, count);
 
-            return GetUnicodeCore(ascii);
+            unsafe
+            {
+                fixed (char* pAscii = ascii)
+                {
+                    return GetUnicodeCore(pAscii, index, count);
```

Comment:
This doesn't look right; ascii is now the substring, so index and count are incorrect.  Is this not being caught by a test?